.github/workflows/test-istio-version.yml

```diff
@@ -20,12 +32,36 @@ jobs:
     outputs:
       target-branch: ${{ inputs.branch_to_test }}
       build-branch: ${{ inputs.branch_to_test }}
-      istio-version: ${{ inputs.istio_version }}
+      istio-version: ${{ steps.determine-istio-version-to-use.outputs.istio-version }}
     steps:
-      # The initialize job gathers variables for later use in jobs.
-      - run: echo "target-branch -> ${{ inputs.branch_to_test }}"
-      - run: echo "build-branch -> ${{ inputs.branch_to_test }}"
-      - run: echo "istio-version -> ${{ inputs.istio_version }}"
+    # The initialize job gathers variables for later use in jobs.
+    - name: Determine Istio Version to use
+      id: determine-istio-version-to-use
+      run: |
+        if [ -z "${{ github.event.inputs.istio_version }}" ]; then
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" && -n "${{ github.event.inputs.istio_minor_version_offset }}" ]]; then
+            OFFSET="${{ github.event.inputs.istio_minor_version_offset }}"
+            # Convert to absolute value - we want a positive offset, though some people might find it more intuitive to say "-1" offset for the previous version
+            OFFSET=$(( OFFSET < 0 ? -OFFSET : OFFSET ))
+          else
+            case "${{ github.event.schedule }}" in
+              "0 2 * * *") OFFSET=0 ;;
```

Comment:
not sure we need to run the latest istio version nightly because we already run it on every PR. Not a big deal, though.

Reply:
I removed the third cron schedule - no longer testing the latest version nightly